### ISSUE TITLE
How do we smoothen User Experience of Github/Gitlab issue/commits collection for both NewUser/Advanced User

### ISSUE THREAD

**Issue by klesh** [merico-dev/lake#1713]
## Description

Problems:
1. Consider the `pipeline` user flow alone: Github/Gitlab both have api rate limit mechanism, so we developed `gitextractor` to collect `commits` and `ref` by cloning repo instead of calling api. However, this approach introduce learning curve to our user, they have to know about why/what/how we have to collect data from a single repo by 2 separate plugin.
2. Consider the big picture: we have beening alway lacking a `wizard` feature to guide user through out setup process, maybe, we could solve the problem above by a step by step guide. and `Wizard` is a must, no matter what, the detail would be discussed in another issue [to put wizard issue link here]

### Solution 1: `gitextractor` as `lib` by @hezyin 
`gitextractor` to provide a set of functions for Github/Gitlab utilization

Questions to be answered:
1. Do we save `commits/refs` into tool layer tables?
2. Do we seperate `gitextractor` subtasks into `commits/branch/tag`?
3. Do these `subtasks` present as a set of `subtasks` in Github/Gitlab plugin
4. Now that Github/Gitlab depend on gitextractor, they have to be updated if gitextractor got updated
5. Subtasks can only be executed in sequetial order, so issues/commits can not be collected in parallel

### Solution 2: `wizard` would create a optmized pipeline for user by @klesh 
all plugins stay separated, and `wizard` ask user sth like `Do u have a github repo`,`is it public` sth like that, and create a optimized pipeline to collect `issues` , `commits` and other in parallel, and prompt user to create a `blueprint` for recurring data collection if pipeline was carried out successfully.

Advantages:
1. No need to refactor `gitextractor`, and do anything specifically for `github/gitlab`
2. Github/Gitlab issues collection and commits collection can run in parallel

## Has the Feature been Requested Before?
No

**Comment by yumengwang03** [comment]
Related to #1700

**Comment by klesh** [comment]
Addressed by #1725
